src/Microsoft.Health.Dicom.Blob/Features/Storage/BlobFileStore.cs

```diff
@@ -58,12 +60,24 @@ public class BlobFileStore : IFileStore
         var blobUploadOptions = new BlobUploadOptions { TransferOptions = _options.Upload };
         stream.Seek(0, SeekOrigin.Begin);
 
+        BlobContentInfo info = null;
+
         await ExecuteAsync(async () =>
         {
-            await blobClient.UploadAsync(stream, blobUploadOptions, cancellationToken);
+            info = await blobClient.UploadAsync(stream, blobUploadOptions, cancellationToken);
         });
 
-        return blobClient.Uri;
+        if (BlobClient.IsExternal)
```

Comment:
Should this check happen within the `BlobFileStore`? My inclination would be for the store to always return the path metadata. Instead, it's up to the orchestrator to pass the information over to the SQL store when completing the STOW operation